File: src/main/java/bad/robot/radiate/teamcity/EnvironmentVariableConfiguration.java
package bad.robot.radiate.teamcity;

import static bad.robot.radiate.Environment.getEnvironmentVariable;
import static java.lang.Integer.valueOf;

public class EnvironmentVariableConfiguration implements TeamCityConfiguration {

    @Override
    public String host() {
        return getEnvironmentVariable("TEAMCITY_HOST");
    }

    @Override
    public Integer port() {
        return valueOf(getEnvironmentVariable("TEAMCITY_PORT", "8111"));
    }

    @Override
    public Iterable<Project> filter(Iterable<Project> projects) {
        return projects; // no filtering
    }

    @Override
    public Password password() {
        return Password.password(getEnvironmentVariable("TEAMCITY_PASSWORD", null));
    }

    @Override
    public Username username() {
        return Username.username(getEnvironmentVariable("TEAMCITY_USER", null));
    }

    @Override
    public Authorisation authorisation() {
        return Authorisation.authorisationFor(username(), password());
    }
}

File: src/main/java/bad/robot/radiate/teamcity/EnvironmentVariableConfiguration.scala
package bad.robot.radiate.teamcity

import bad.robot.radiate.Environment.getEnvironmentVariable
import java.lang.Integer.valueOf

class EnvironmentVariableConfigurationS extends TeamCityConfigurationS {
  def host = getEnvironmentVariable("TEAMCITY_HOST")

  def port = valueOf(getEnvironmentVariable("TEAMCITY_PORT", "8111"))

  def filter(projects: Iterable[ProjectScala]) = projects

  def password = PasswordS.password(getEnvironmentVariable("TEAMCITY_PASSWORD", null))

  def username = UsernameS.username(getEnvironmentVariable("TEAMCITY_USER", null))

  def authorisation = AuthorisationS.authorisationFor(username, password)
}